### PR TITLE
feat(inputs.kube_inventory): Support filtering pods and nodes by node name

### LINKED ISSUE #13878: Filter kuberetes_inventory input plugin nodes and pods by node name
### Use Case

When this plugin is installed on large clusters it creates stress on the API server since it requests information on all of the pods & nodes in the cluster
To support scaling we want to install it on a daemonset 
But without the option to filter by node name it will still get all the information for each pod in the daemon and create an even bigger pressure on the API server

### Expected behavior

Configure metrics on pods & nodes by node name on a daemon and get the relevant information with good performance 

### Actual behavior

A lot of pressure on the API server and timeout failures to get inventory date 

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by Noy-Simon**
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

[resolves #13878](https://github.com/influxdata/telegraf/issues/13878)

Add the option to filter nodes and pods by node name in the kuberetes_inventory input plugin
Add an option to get pods info using Kubelet to avoid loading on the API server

### CODE REVIEW

**Comment by Noy-Simon** [comment]
!signed-cla

**Comment by Noy-Simon** [comment]
signed-cla!

**Comment by Noy-Simon** [comment]
signed-cla

**Comment by Noy-Simon** [comment]
signed-cla!

**Comment by Noy-Simon** [comment]
!signed-cla

**Comment by srebhan** [comment]
!signed-cla

File: plugins/inputs/kube_inventory/README.md [hunk 1]
```diff
@@ -184,6 +186,7 @@ tls_key = "/run/telegraf-kubernetes-key"
 ## Metrics
 
 - kubernetes_daemonset
+
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/README.md:189]: Please remove those newlines. Same for the others below.

File: plugins/inputs/kube_inventory/client.go [hunk 2]
```diff
@@ -86,10 +87,14 @@ func (c *client) getIngress(ctx context.Context) (*netv1.IngressList, error) {
...
 
-func (c *client) getNodes(ctx context.Context) (*corev1.NodeList, error) {
+func (c *client) getNodes(ctx context.Context, name string) (*corev1.NodeList, error) {
 	ctx, cancel := context.WithTimeout(ctx, c.timeout)
 	defer cancel()
-	return c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if name == "" {
+		return c.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	}
+	fieldSelector := fmt.Sprintf("metadata.name=%s", name)
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/client.go:96]: How about
```suggestion
	var fieldSelector string
	if name != "" {
		fieldSelector = "metadata.name="+name
	}
```

File: plugins/inputs/kube_inventory/client.go [hunk 3]
```diff
@@ -104,10 +109,14 @@ func (c *client) getPersistentVolumeClaims(ctx context.Context) (*corev1.Persist
...
-func (c *client) getPods(ctx context.Context) (*corev1.PodList, error) {
+func (c *client) getPods(ctx context.Context, nodeName string) (*corev1.PodList, error) {
 	ctx, cancel := context.WithTimeout(ctx, c.timeout)
 	defer cancel()
-	return c.CoreV1().Pods(c.namespace).List(ctx, metav1.ListOptions{})
+	if nodeName == "" {
+		return c.CoreV1().Pods(c.namespace).List(ctx, metav1.ListOptions{})
+	}
+	fieldSelector := fmt.Sprintf("spec.nodeName=%s", nodeName)
+	return c.CoreV1().Pods(c.namespace).List(ctx, metav1.ListOptions{FieldSelector: fieldSelector})
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/client.go:119]: Same as above.

File: plugins/inputs/kube_inventory/sample.conf [hunk 4]
```diff
@@ -7,6 +7,9 @@
   ## Namespace to use. Set to "" to use all namespaces.
   # namespace = "default"
 
+  ## Node name to filter to.
+  # node = "minikube"
```
Comment (srebhan) [inline plugins/inputs/kube_inventory/sample.conf:11]: The commented value should reflect the default. So I would suggest
```suggestion
  ## Node name to filter to. No filtering by default.
  # node = ""
```

**Review by srebhan — COMMENTED** [review]
Thanks @Noy-Simon for the PR! Some minor comments...

**Comment by powersj** [comment]
!signed-cla

**Comment by powersj** [comment]
@Noy-Simon you GitHub username is capitalized, but in the form you did not have it capitalized

**Pull request closed without merging**